specification/workloads/resource-manager/Microsoft.Workloads/preview/2021-12-01-preview/monitors.json

```diff
@@ -0,0 +1,1190 @@
+        "x-ms-examples": {
+          "Create a SAP Monitor": {
+            "$ref": "./examples/workloadmonitor/monitors_Create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/monitorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Created Monitor",
+            "schema": {
+              "$ref": "#/definitions/Monitor"
+            }
+          },
+          "201": {
+            "description": "Creating Monitor",
+            "schema": {
+              "$ref": "#/definitions/Monitor"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Monitors"
+        ],
+        "operationId": "monitors_Delete",
+        "x-ms-long-running-operation": true,
+        "summary": "Deletes a SAP monitor.",
+        "description": "Deletes a SAP monitor with the specified subscription, resource group, and monitor name.",
+        "x-ms-examples": {
+          "Deletes a SAP monitor": {
+            "$ref": "./examples/workloadmonitor/monitors_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/monitorNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Delete Monitor",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/OperationStatusResult"
+            }
+          },
+          "202": {
+            "description": "Deleting Monitor"
+          },
+          "204": {
+            "description": "No Monitor to delete"
+          },
```

Comment:
Do you need 200 and 204 for your async delete, 202 + location header and the location header can return 200 and 204. See : [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously)

Reply:
Yes, we are using 200 and 204 in our code for responses depending on certain scenarios like monitor not present on managed rg not present, etc and we are passing this value directly to RPaaS. Resolving the issue for now. will reopen if further discussion required

Reply:
It is up to service to make this a long running or a synchronous api call, rpaas would follow the response sent by service (200 -> sync, else async).
Kept both options in swagger so that service can decide at runtime if operation is synchronized or async, and rpaas can act accordingly

Reply:
So deferring the decision on whether to make an API invocation sync vs async is an anti-pattern as per Azure API guidelines and must be avoided. But I do see why you are doing this